CapybaraLearning/spec/dashboard_spec.rb

```diff
@@ -2,47 +2,51 @@
 
 describe 'When testing demo.opensourcecms website' do
   let(:count_of_ref) { 41 }
-  let(:count_of_buttons) { 9 }
-  let(:count_of_headings) { 6 }
+  let(:count_of_buttons) { 7 }
+  let(:count_of_headings) { 12 }
   let(:count_of_inputs) { 2 }
-  let(:count_of_text) { 47 }
+  let(:count_of_text) { 43 }
+  let(:login_page) { LoginPage.new }
+  let(:home_page) { HomePage.new }
 
   before(:all) do
     Capybara.configure do |config|
       config.run_server = false
       config.default_driver = :poltergeist
     end
-      @login_page = LoginPage.new
-      @home_page = HomePage.new
   end
 
   it 'Verify navigating to the login page' do
-    @login_page.load
+    login_page.load
   end
 
   it 'Verify loggining' do
-    @home_page = @login_page.log_in('opensourcecms', 'opensourcecms')
-    expect(@home_page.loggined?).to be true
+    home_page = login_page.log_in('opensourcecms', 'opensourcecms')
+    expect(home_page.loggined?).to be true
   end
 
   it 'Verify count of links on dashbord page' do
-    expect(@home_page.get_main_page_ref.count).to eq count_of_ref
+    expect(home_page.get_main_page_ref.count).to eq count_of_ref
   end
 
   it 'Verify count of bunnons on dashbord page' do
-    expect(@home_page.get_main_page_buttons.count).to eq count_of_buttons
+    expect(home_page.get_main_page_buttons.count).to eq count_of_buttons
   end
 
   it 'Verify count of headings on dashbord page' do
-    expect(@home_page.get_main_page_headings.count).to eq count_of_headings
+    expect(home_page.get_main_page_headings.count).to eq count_of_headings
+  end
+
+  it 'Verify text of headings on dashbord page' do
+    expect(home_page.correct_headings?).to be true
```

Comment:
thi approach will work, but you will not get the understandable reason, which headings are wrong, better not use correct_headings? method, and just check if names are the same right here in the test. For this you will need to move headings from dashboard_headings method into variable.